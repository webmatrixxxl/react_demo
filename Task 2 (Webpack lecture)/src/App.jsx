import * as React from 'react';

export class App extends React.PureComponent {
  render() {
    return (
      <main>
      	Hi
      </main>
    );
  }
}
