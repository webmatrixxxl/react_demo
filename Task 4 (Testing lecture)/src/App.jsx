import * as React from 'react'

import { Logo } from './components/logo';
import { Header } from './components/header/header';

export class App extends React.PureComponent {
  constructor() {
    super();
    this.state = {
      hasError: null
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: error});
  }

  render() {
    if (this.state.hasError) {
      return (
        <React.Fragment>
          <div>{ this.state.hasError }</div>
        </React.Fragment>
      );
    } else {
      return (
        <React.Fragment>
          <Header>
            <Logo />
          </Header>
          <main></main>
          <footer>
            <Logo />
          </footer>
        </React.Fragment>
      );
    }
  }
}
