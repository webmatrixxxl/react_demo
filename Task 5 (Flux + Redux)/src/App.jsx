import * as React from 'react';
import { Provider } from 'react-redux';

import { Logo } from './components/logo';
import { Header } from './components/header/header';

import Post from './components/Post';

import store from 'store';


export class App extends React.PureComponent {
  constructor() {
    super();
    this.state = {
      hasError: null
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: error});
  }

  render() {
    if (this.state.hasError) {
      return (
        <Provider store={store}>
          <React.Fragment>
            <div>{ this.state.hasError }</div>
          </React.Fragment>
        </Provider>
      );
    } else {
      return (
        <Provider store={store}>
          <React.Fragment>
            <Post />
            <Header>
              <Logo />
            </Header>
            <main></main>
            <footer>
              <Logo />
            </footer>
           </React.Fragment>
        </Provider>
      );
    }
  }
}
