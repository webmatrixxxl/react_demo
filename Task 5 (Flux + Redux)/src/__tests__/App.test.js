import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme'
import { shallow, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { App } from './../App';

// jest.mock('./../components', () => 'Component');

describe('App', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<App />).toJSON();
    expect(component).toMatchSnapshot();
  })


  it('show error', () => {
    const component = shallow(<App />);
    component.setState({ hasError: 'test error' });
    expect(component.state('hasError')).toEqual('test error');
  })
})
