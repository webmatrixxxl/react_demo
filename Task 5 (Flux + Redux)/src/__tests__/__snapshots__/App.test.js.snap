// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
Array [
  <header
    style={
      Object {
        "backgroundImage": "url(./bg--netflix.jpg)",
      }
    }
  >
    <div>
      <h1
        className="header__logo"
      >
        netflixroulette
      </h1>
      <button>
        Search
      </button>
    </div>
    <div
      className="search"
    >
      <h3>
        find your movie
      </h3>
      <div>
        <input
          name="searchMovie"
          type="search"
        />
         
        <button>
          i
        </button>
      </div>
      <div>
        <div>
          search by 
          <button>
            title
          </button>
           
          <button>
            genre
          </button>
        </div>
        <button>
          search
        </button>
      </div>
    </div>
  </header>,
  <main />,
  <footer>
    <h1
      className="header__logo"
    >
      netflixroulette
    </h1>
  </footer>,
]
`;
